connection.go

```diff
@@ -608,14 +608,27 @@ func (c *Connection) readState() connectionState {
 // incoming frame to a channel; the init handlers are a notable exception,
 // since we cannot process new frames until the initialization is complete.
 func (c *Connection) readFrames(_ uint32) {
+	headerBuf := make([]byte, FrameHeaderSize)
+
+	handleErr := func(err error) {
+		if c.closeNetworkCalled.Load() == 0 {
+			c.connectionError("read frames", err)
+		} else {
+			c.log.Debugf("Ignoring error after connection was closed: %v", err)
```

Comment:
does this need to be a closure? it doesn't appear to be accessing anything outside.

Reply:
doesn't need to be a closure, but wanted to avoid adding yet another method to `*Connection`, this logic is very local to `readFrames`, this was just broken out into a function to DRY up the body.